### PR TITLE
Add support for revert a commit

### PR DESCRIPTION
**Pull request by mtojek**
Hi,

the client library lacks support for "revert a commit" operation, that Gitlab introduced in 11.5 version. This PR allows for such operation.

/cc @svanharmelen

### CODE REVIEW

File: commits.go [hunk 1]
```diff
@@ -492,7 +492,7 @@ type CherryPickCommitOptions struct {
 	TargetBranch *string `url:"branch" json:"branch,omitempty"`
 }
 
-// CherryPickCommit sherry picks a commit to a given branch.
+// CherryPickCommit cherry picks a commit to a given branch.
```
Comment (svanharmelen) [inline commits.go]: 👍

File: commits.go [hunk 2]
```diff
@@ -515,3 +515,34 @@ func (s *CommitsService) CherryPickCommit(pid interface{}, sha string, opt *Cher
 
 	return c, resp, err
 }
+
+// RevertCommitOptions represents the available options for revert a commit.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/commits.html#revert-a-commit
+type RevertCommitOptions struct {
+	TargetBranch *string `url:"branch" json:"branch,omitempty"`
```
Comment (svanharmelen) [inline commits.go]:
```suggestion
	Branch *string `url:"branch,omitempty" json:"branch,omitempty"`
```

File: commits.go [hunk 3]
```diff
@@ -515,3 +515,34 @@ func (s *CommitsService) CherryPickCommit(pid interface{}, sha string, opt *Cher
 
 	return c, resp, err
 }
+
+// RevertCommitOptions represents the available options for revert a commit.
```
Comment (svanharmelen) [inline commits.go]:
```suggestion
// RevertCommitOptions represents the available RevertCommit() options.
```

File: commits.go (hunk 2 above)
Comment (mtojek) [reply commits.go]: Done

File: commits.go (hunk 3 above)
Comment (mtojek) [reply commits.go]: Done

**Comment by mtojek** [comment]
> I made a few suggestions that I like to see changed before we merge this one. Additionally I noticed that the `CherryPickCommitOptions` (and related) code actually needs the same changes. Feel free to update them to be consistent, but I can do that as well if you prefer that. Thanks!

Thank you for the guidance, Sander. I've updated `CherryPickCommitOptions` as well. Feel free to review and merge if ready.

**Comment by svanharmelen** [comment]
Thanks @mtojek! LGTM!

**Comment by mtojek** [comment]
Thank you too!

**Pull request merged**